kernel/thor/generic/service.cpp

```
@@ -574,16 +574,17 @@ namespace posix {
 					continue;
 				}
 
-				if(!(req->flags() & 4)) // MAP_FIXED.
-					assert(!"TODO: implement non-fixed mappings");
```

It's good to get rid of the assertion here, but shouldn't we return an error for unsupported flags?